<launch>
  <group ns="rtabmap">

    <node pkg="nodelet" type="nodelet" name="rgbd_sync" args="standalone rtabmap_ros/rgbd_sync" output="screen">
      <remap from="rgb/image"        to="/camera/rgb/image_rect_color"/>
      <remap from="depth/image"      to="/camera/depth_registered/image_raw"/>
      <remap from="rgb/camera_info"  to="/camera/rgb/camera_info"/>
      <remap from="rgbd_image"       to="rgbd_image"/> <!-- output -->
      
      <param name="approx_sync"       value="false"/> 
    </node>

    <!-- Odometry -->
    <node pkg="rtabmap_ros" type="rgbd_odometry" name="rgbd_odometry" output="screen">
      <param name="subscribe_rgbd" type="bool"   value="true"/>
      <param name="frame_id"       type="string" value="base_link "/>     
      <param name="Odom/ResetCountdown" value="10" />

      <remap from="rgbd_image" to="rgbd_image"/>
    </node>

    <node name="rtabmap" pkg="rtabmap_ros" type="rtabmap" output="screen" args="--delete_db_on_start">
          <param name="frame_id"        type="string" value="base_link"/>
          <param name="subscribe_depth" type="bool"   value="false"/>
          <param name="subscribe_rgbd"  type="bool"   value="true"/>
          <param name="rtabmapviz" type="bool" value="true"/>

          <remap from="odom"       to="odom"/>
          <remap from="rgbd_image" to="rgbd_image"/>

          <param name="queue_size"  type="int"  value="10"/>
          <param name="approx_sync" type="bool" value="false"/>

          <!-- RTAB-Map's parameters
            https://github.com/introlab/rtabmap/blob/master/corelib/include/rtabmap/core/Parameters.h
           -->
          <param name="RGBD/AngularUpdate"        type="string" value="0.01"/>
          <param name="RGBD/LinearUpdate"         type="string" value="0.01"/>
          <param name="RGBD/OptimizeFromGraphEnd" type="string" value="false"/>
          <param name="RGBD/StartAtOrigin"        type="bool" value="true"/>
          <param name="Odom/Strategy"             type="int" value="0"/>
          <param name="Rtabmap/StartNewMapOnLoopClosure" value="false"/>
          <param name="Vis/MaxFeatures" type="string" value="600"/>
    </node>


    <!--
    <node name="rtabmapviz" pkg="rtabmap_ros" type="rtabmapviz" output="screen">
      <param name="subscribe_rgbd"  type="bool" value="true"/>
    </node>  
    -->
    
        
  </group>
</launch>